include/deal.II/meshworker/copy_data.h

```diff
@@ -0,0 +1,144 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2019 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_meshworker_copy_data_h
+#define dealii_meshworker_copy_data_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/types.h>
+
+#include <deal.II/lac/full_matrix.h>
+#include <deal.II/lac/vector.h>
+
+#include <vector>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace MeshWorker
+{
+  /**
+   * Helper copy data struct.
+   *
+   * This class is a good default drop in CopyData object for the
+   * WorkStream::run() and MeshWorker::mesh_loop() functions.
+   *
+   * It allows you to store arbitrary number of (local) FullMatrices, vectors,
+   * and local_dof_indices vectors.
+   *
+   * @author Luca Heltai, 2019.
+   */
+  template <int n_matrices    = 1,
+            int n_vectors     = n_matrices,
+            int n_dof_indices = n_matrices>
+  struct CopyData
+  {
+    /**
+     * Initialize everything with the same @p size.
+     */
+    CopyData(const unsigned int size);
+
+    /**
+     * For every object, specify the size they should have.
+     */
+    CopyData(
+      const std::array<std::array<unsigned int, 2>, n_matrices> &matrix_sizes,
+      const std::array<unsigned int, n_vectors> &                vector_sizes,
+      const std::array<unsigned int, n_dof_indices> &dof_indices_sizes);
+
+    /**
+     * Deep copy constructor.
+     */
+    CopyData(const CopyData<n_matrices, n_vectors, n_dof_indices> &other) =
+      default;
+
+    /**
+     * Allow resetting of all elements of the struct to zero, by simply
+     * calling (*this) = 0;
```

Comment:
`calling <code>(*this) = 0;</code>.`

Reply:
Maybe its worth mentioning that using anything other than `=0` will result in an error (which I guess is thrown on the resetting of the first matrix).